File: client/bcephal-client-model/Profiles/ProfileEditorData.cs
using Bcephal.Models.Base;
using Bcephal.Models.Clients;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Profiles
{
    public class ProfileEditorData : EditorData<Profile>
    {

        public ObservableCollection<ClientFunctionality> Functionalities { get; set; }

        public ObservableCollection<Nameable> Users { get; set; }


        public ObservableCollection<Nameable> Roles { get; set; }

    }
}

File: server/bcephal-security-service/src/main/java/com/moriset/bcephal/security/domain/ProfileEditorData.java
/**
 * 
 */
package com.moriset.bcephal.security.domain;

import java.util.ArrayList;
import java.util.List;

import com.moriset.bcephal.domain.EditorData;
import com.moriset.bcephal.domain.Nameable;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Moriset
 *
 */
@Data
@EqualsAndHashCode(callSuper=false)
public class ProfileEditorData extends EditorData<Profile> {

	private List<Nameable> users;
	private List<Nameable> roles;
	private List<ClientFunctionality> functionalities;	
	private List<ProfileType> types;

	public ProfileEditorData() {
		types = new ArrayList<>(0);
		users = new ArrayList<>(0);
		roles = new ArrayList<>(0);
		functionalities = new ArrayList<>(0);
	}
	
}
